lib/interceptor/decompress.js

```diff
@@ -0,0 +1,188 @@
+'use strict'
+
+const { createInflate, createGunzip, createBrotliDecompress, createZstdDecompress } = require('node:zlib')
+const { pipeline } = require('node:stream')
+const DecoratorHandler = require('../handler/decorator-handler')
+
+class DecompressHandler extends DecoratorHandler {
+  #decompressors = []
+  #pipelineStream = null
+  #skipStatusCodes
+  #skipErrorResponses
+
+  constructor (handler, { skipStatusCodes = [204, 304], skipErrorResponses = true } = {}) {
+    super(handler)
+    this.#skipStatusCodes = skipStatusCodes
+    this.#skipErrorResponses = skipErrorResponses
+  }
+
+  #shouldSkipDecompression (contentEncoding, statusCode) {
+    if (!contentEncoding || statusCode < 200) return true
+    if (this.#skipStatusCodes.includes(statusCode)) return true
+    if (this.#skipErrorResponses && statusCode >= 400) return true
+    return false
+  }
+
+  #createDecompressor (encoding) {
+    const supportedEncodings = {
+      gzip: createGunzip,
+      'x-gzip': createGunzip,
+      br: createBrotliDecompress,
+      zstd: createZstdDecompress,
```

Comment:
If this is only supported on `>v22`, let's conditionally import it; otherwise can lead to problems if used on v20

Reply:
I can trigger flows manually, no problerm